js_modules/dagster-ui/packages/ui-core/src/app/QueryRefresh.tsx

```diff
@@ -45,6 +46,9 @@ export function useQueryRefreshAtInterval(
   const queryResultRef = React.useRef(queryResult);
   queryResultRef.current = queryResult;
 
+  const customRefetchRef = React.useRef(customRefetch);
+  customRefetchRef.current = customRefetch;
```

Comment:
I'm a little confused by setting the `current` value during render. Why is a ref needed?

Reply:
a year late but the ref keeps track of the current callback. Basically we have a useEffect that uses this value (eg. when the document becomes visible again after tabbing away) to refetch the data. If we added `customRefetch` directly as a dependency on that effect, AND the caller is not using `React.useCallback` then an infinite loop could happen. This basically circumvents that footgun. It's also better perf because we don't go the useEffect unsubscribe and resubscribe loop.